### PR TITLE
Revisions oct archit

### PR DESCRIPTION
**Pull request by NeoChaos12**
Addresses the following issues:
#35 
#45 
#46 
#49 
#50 
#51

### CODE REVIEW

File: hpolib/abstract_benchmark.py [hunk 1]
```diff
@@ -154,13 +163,17 @@ def wrapper(self, configuration: Union[np.ndarray, ConfigSpace.Configuration, Di
             # If kwargs contains the 'fidelity' arg, extract any fidelity parameters it contains and fill in
             # default values for the rest.
             default_fidelities = self.fidelity_space.get_default_configuration()
+            fidelity = dict(fidelity)
```
Comment (PhMueller) [inline hpolib/abstract_benchmark.py:166]: this does not work if fidelity is ``None``

File: hpolib/abstract_benchmark.py [hunk 2]
```diff
@@ -125,7 +131,7 @@ def wrapper(self, configuration: Union[np.ndarray, ConfigSpace.Configuration, Di
                                 f'was {type(configuration)}')
 
             self.configuration_space.check_configuration(config)
-            return foo(self, configuration, **kwargs)
+            return foo(self, configuration=config.get_dictionary(), **kwargs)
```
Comment (PhMueller) [inline hpolib/abstract_benchmark.py:134]: See discussion in the open issue #45

File: setup.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def read_file(file_name):
     version=read_file('hpolib/__version__.py').split()[-1].strip('\''),
     packages=setuptools.find_packages(exclude=['*.tests', '*.tests.*',
                                                'tests.*', 'tests'],),
-    python_requires='>=3.6, <3.8',
+    python_requires='>=3.6',
```
Comment (PhMueller) [inline setup.py:39]: have you tested it with a version > 3.8? I can remember, that there was a problem with some dependencies. But I am not sure if it is still the case. Maybe we should add another travis configuration with python3.8.X.

File: extra_requirements/svm.json [hunk 4]
```diff
@@ -1,3 +1,3 @@
 {
-  "svm": ["pandas>=0.22.2,<0.24.2","openml==0.10.2","scikit-learn>=0.18.1"]
```
Comment (PhMueller) [inline extra_requirements/svm.json:2]: We encountered a problem when having openml==0.10.2 and Pandas >24.2. Before we can allow those versions (or a newer openml version, which currently is required for the openml tasks), we need to fix this.

Comment (PhMueller) [reply extra_requirements/svm.json:2]: see issue #51

**Comment by PhMueller** [comment]
I would suggest to set the target of this PR to development and then to master!

**Review by PhMueller — CHANGES_REQUESTED** [review]
I would suggest to set the target of this PR to development and then to master!

File: setup.py (hunk 3 above)
Comment (NeoChaos12) [reply setup.py:39]: I think Python 3.8's early versions have a bit of a rough history and were subsequently patched up. 3.8.3 was released in May and I've been using it for a while without any issues.

**Pull request closed without merging**